test/db/index.js

```diff
@@ -0,0 +1,449 @@
+const chai = require('chai');
+const chaiExclude = require('chai-exclude');
+
+chai.use(chaiExclude);
+const should = chai.should();
+const { expect } = chai;
+const fs = require('fs-extra');
+const { initDB, db } = require('../../src/db');
+const { getDbDir } = require('../../src/config');
+const DBHelper = require('../../src/db/db-helper');
+const { EVENT_STATUS } = require('../../src/constants');
+
+describe('db', () => {
+  before(async () => {
+    await initDB();
+  });
+
+  /* Blocks */
+  describe('test block db', () => {
+    describe('insert', () => {
+      let blockCount;
+      let blocks;
+      const toBeInsertedBlockNum = 11111;
+      const toBeInsertedBlockTime = Date.now();
+      it('should start with empty block db', async () => {
+        blockCount = await db.Blocks.count({});
+        blockCount.should.equal(0);
+      });
+
+      it('should insert with new block num', async () => {
+        await DBHelper.insertBlock(toBeInsertedBlockNum, toBeInsertedBlockTime);
+        blocks = await db.Blocks.find({});
+        blocks.length.should.equal(1);
+        blocks[0].blockNum.should.equal(toBeInsertedBlockNum);
+        blocks[0].blockTime.should.equal(toBeInsertedBlockTime);
+      });
+
+      it('should not insert with existed block num', async () => {
+        const toBeInsertedFailBlockTime = Date.now();
+        await DBHelper.insertBlock(toBeInsertedBlockNum, toBeInsertedFailBlockTime);
+        blocks = await db.Blocks.find({});
+        blocks.length.should.equal(1);
+        blocks[0].blockNum.should.equal(toBeInsertedBlockNum);
+        blocks[0].blockTime.should.equal(toBeInsertedBlockTime);
+      });
+    });
+
+    describe('find latest block', () => {
+      let blockCount;
+      it('find latest on empty block db', async () => {
+        db.Blocks.remove({}, { multi: true });
```

Comment:
dont need to change it here but for future reference. if you are doing tests which require having a constant state (db in these cases), a good practice is to optimize the test order itself.

for example, for the Blocks db, you start with all your tests which require an empty db first. then you can do all the tests which check for existing entries in the db after you add them in. that way you don't have to keep clearing the db for the empty tests. it's like following a lifecycle for the test db.